src/client.ml

```diff
@@ -1,10 +1,30 @@
 type t =
-  { api_key : string
-  ; gen_url : string -> string
+  { mutable api_key : string
+  ; mutable url : string
   ; c : Curl.t
+  ; mutable model : string
   }
 
-let create api_key =
+let create  api_key=
   let base_url = "https://api.openai.com" in
-  { api_key; gen_url = ( ^ ) base_url; c = Ezcurl_lwt.make () }
+  {
+    api_key
+  ; url = base_url
+  ; c = Ezcurl_lwt.make ()
+  ; model="no-model"
+  }
+;;
+let create_init =
+  let base_url = "" in
+  let model = "" in
+  let api_key = "" in
+  {
+    api_key
+  ; url = base_url
+  ; c = Ezcurl_lwt.make ()
+  ; model
+  }
```

Comment:
It is preferred to `create` for each models than initiate empty object and update.